pages/events/[id].vue

```diff
@@ -0,0 +1,21 @@
+<script setup lang="ts">
+
+import Event from "~/components/events/Event.vue"
+
+const route = useRoute()
+let id: string;
+
+if (route.params.id == undefined) {
```

Comment:
never use `==` in js, only `===`

it might be worth considering the following:
```ts
const route = useRoute()
const id = route.params.id as unknown as string
```
the router will guarantee that the id is defined and of type string, as it won't navigate there if it isn't